src/sql/workbench/contrib/dashboard/browser/core/dashboardPage.component.ts

```diff
@@ -126,6 +151,48 @@ export abstract class DashboardPage extends AngularDisposable implements IConfig
 			};
 			this.createTabs(tempWidgets);
 		}
+
+		this.createTaskbar(this.taskContainer.nativeElement);
+	}
+
+	protected createTaskbar(parentElement: HTMLElement): void {
+		let taskbarContainer = DOM.append(parentElement, DOM.$('div'));
+		this.taskbar = this._register(new Taskbar(taskbarContainer));
+		this._restoreAction = new RestoreToolbarAction(this.restore, this);
```

Comment:
What if a dashboard doesn't want all these actions (such as the BDC dashboard)? Ideally these actions should be contributed through a more generic means that each dashboard type sets itself.

Reply:
yes, the toolbar needs to be able to have different items per page.